src/main/java/com/google/codeu/servlets/SignUpServlet.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.codeu.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.JsonObject;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.lang.Integer;
+import com.google.codeu.data.User;
+import com.google.codeu.data.User.AccountType;
+import com.google.codeu.data.Datastore;
+
+
+
+@WebServlet("/signup")
+public class SignUpServlet extends HttpServlet {
+
+  private Datastore datastore;
+
+  public static final int NULL_EMAIL_ERROR = 1; // email cannot be null
+  public static final int NULL_PASSWORD_ERROR = 2; // password cannot be null
+  public static final int NULL_ACCCOUNT_TYPE_ERROR = 3; //both account types cannot be selected
+  public static final int BOTH_ACCCOUNT_TYPE_ERROR = 4; //at least one account types should be selected
+  public static final int INVALID_PASSWORD_ERROR = 5; // used in login servlet
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+
+    request.setAttribute("error", false);
```

Comment:
This looks like a decent amount of duplicate code shared with LoginServlet. It might be worth refactoring into a shared utility method.

Up to you. Since the code is only done twice, it's still relatively easy to maintain.